2.0/m_conn_vhost.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2017 genius3000 <genius3000@g3k.solutions>>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: genius3000 */
+/* $ModAuthorMail: genius3000@g3k.solutions */
+/* $ModDesc: Adds a "vhost" option to connect blocks */
+/* $ModDepends: core 2.0 */
+/* $ModConfig: Within connect block: vhost="vhost.here" */
+
+
+#include "inspircd.h"
+
+class ModuleVhostOnConnect : public Module
+{
+ public:
+	void init()
+	{
+		ServerInstance->Modules->Attach(I_OnUserConnect, this);
+	}
+
+	void Prioritize()
+	{
+		/* Let's go after conn_umodes in case +x also gets set */
+		Module* connumodes = ServerInstance->Modules->Find("m_conn_umodes.so");
+		ServerInstance->Modules->SetPriority(this, I_OnUserConnect, PRIORITY_AFTER, connumodes);
+	}
+
+	virtual Version GetVersion()
+	{
+		return Version("Sets a connect block configured vhost on users when they connect");
+	}
+
+	virtual void OnUserConnect(LocalUser* user)
+	{
+		ConfigTag* tag = user->MyClass->config;
+		std::string vhost = tag->getString("vhost");
+
+		if (!vhost.empty())
```

Comment:
Not a bad idea, I was going for the "less is more" style with the simplicity here but ensuring we don't crash the server on over-length or an invalid character is probably a good plan.